go/ioutil2/ioutil.go

```diff
@@ -20,11 +20,16 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
 		return err
 	}
 	n, err := f.Write(data)
+	if err == nil {
```

Comment:
Yes, you have both made some good comments. Let me think about how to
resolve some of these issues, the error handling really isn't good enough
here. The function is better than it was, but not correct.

On Sat, Mar 21, 2015 at 1:18 PM, sougou <notifications@github.com> wrote:

> In go/ioutil2/ioutil.go
> <https://github.com/youtube/vitess/pull/510#discussion_r26896454>:
>
> > @@ -20,11 +20,16 @@ func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
> >  		return err
> >  	}
> >  	n, err := f.Write(data)
> > +	if err == nil {
>
> I went back and read the doc for syncfs & close. It says that close does
> not have to sync. But I still think it's safe. I think the importance of
> syncfs is only relevant for data loss due to power failure scenarios.
> However, they say you have to check the return value of f.Close because an
> error there means that a previous write could have failed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/youtube/vitess/pull/510/files#r26896454>.
>